pkgs/tools/system/uefitool/new-engine.nix

```
@@ -0,0 +1,38 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  qtbase,
+  cmake,
+  wrapQtAppsHook,
+  zip,
+}:
+
+stdenv.mkDerivation rec {
```

```suggestion
stdenv.mkDerivation (finalAttrs: {
```

rec is discouraged/deprecated in nixpkgs, the finalAttrs pattern is the current convention.

---

I applied this for `new-engine.nix` as suggested, but left `rec` for `old-engine.nix`. The latter gets the `mkDerivation` filled in from `libsForQt5.callPackage` which appears to not support the lambda / `finalAttrs` variant of the invocation. I can use a `let` instead, if that is preferred over using `rec`?

---

Don't use `let ... in`, that one is terrible for overrides. `rec` is still fine if finalAttrs is unavailable.